### PR TITLE
add unit test for function AsScaledInt64

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
add unit test for function AsScaledInt64
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by thockin** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/60790)

File: staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go [hunk 1]
```diff
@@ -131,3 +131,28 @@ func TestAmountSign(t *testing.T) {
...
+
+func TestInt64AmountAsScaledInt64(t *testing.T) {
+	for _, test := range []struct {
+		value  int64
+		scale  Scale
+		scaled Scale
+		result int64
+		ok     bool
+	}{
+		{100, 0, 5, 1, true},
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go]: Name the struct values.  Don't rely on the struct order.

```go
{
   value: 100,
   scale: 0,
```

Also, put a `name` in each test case and use the name in a `test.Run(test, func(t *testing.T){})` call

Comment (hanxiaoshuai) [reply staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go]: @deads2k Done

File: staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go [hunk 2]
```diff
@@ -131,3 +131,30 @@ func TestAmountSign(t *testing.T) {
...
+		{"test when i.scale < scaled ", int64Amount{value: 100, scale: 0}, 5, 1, true},
+		{"test when i.scale = scaled", int64Amount{value: 100, scale: 1}, 1, 100, true},
+		{"test when i.scale > scaled and result doesn't overflow", int64Amount{value: 100, scale: 5}, 2, 100000, true},
+		{"test when i.scale > scaled and result overflows", int64Amount{value: 876, scale: 30}, 4, 0, false},
+		{"test when i.scale < 0 and fraction exists", int64Amount{value: 93, scale: -1}, 0, 10, true},
+		{"test when i.scale < 0 and fraction doesn't exist", int64Amount{value: 100, scale: -1}, 0, 10, true},
+		{"test when i.value < 0 and fraction exists", int64Amount{value: -1932, scale: 2}, 4, -20, true},
+		{"test when i.value < 0 and fraction doesn't exists", int64Amount{value: -1900, scale: 2}, 4, -19, true},
+	} {
+		r, ok := test.i.AsScaledInt64(test.scaled)
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go]: Use the subtest feature described here: https://blog.golang.org/subtests to allow the tests to easily fail individually.

Comment (hanxiaoshuai) [reply staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go]: @deads2k Done, is it like this?

File: staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go [hunk 3]
```diff
@@ -131,3 +132,32 @@ func TestAmountSign(t *testing.T) {
...
+		{"test when i.scale < scaled ", int64Amount{value: 100, scale: 0}, 5, 1, true},
+		{"test when i.scale = scaled", int64Amount{value: 100, scale: 1}, 1, 100, true},
+		{"test when i.scale > scaled and result doesn't overflow", int64Amount{value: 100, scale: 5}, 2, 100000, true},
+		{"test when i.scale > scaled and result overflows", int64Amount{value: 876, scale: 30}, 4, 0, false},
+		{"test when i.scale < 0 and fraction exists", int64Amount{value: 93, scale: -1}, 0, 10, true},
+		{"test when i.scale < 0 and fraction doesn't exist", int64Amount{value: 100, scale: -1}, 0, 10, true},
+		{"test when i.value < 0 and fraction exists", int64Amount{value: -1932, scale: 2}, 4, -20, true},
+		{"test when i.value < 0 and fraction doesn't exists", int64Amount{value: -1900, scale: 2}, 4, -19, true},
+	} {
+		t.Run(fmt.Sprintf("%s", test.name), func(t *testing.T) {
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/api/resource/amount_test.go]: sprintf isn't needed.

**Comment by hanxiaoshuai** [comment]
cc @deads2k @caesarxuchao

**Pull request merged**